hugr-model/src/v0/text/parse.rs

```diff
@@ -697,6 +695,44 @@ impl<'a> ParseContext<'a> {
             unreachable!("expected a symbol");
         }
     }
+
+    fn parse_string(&self, token: Pair<'a, Rule>) -> ParseResult<&'a str> {
+        assert_eq!(token.as_rule(), Rule::string);
+
+        // Any escape sequence is longer than the character it represents.
+        // Therefore the length of this token (minus 2 for the quotes on either
+        // side) is an upper bound for the length of the string.
+        let capacity = token.as_str().len() - 2;
+        let mut string = BumpString::with_capacity_in(capacity, self.bump);
+        let tokens = token.into_inner();
+
+        for token in tokens {
+            match token.as_rule() {
+                Rule::string_raw => string.push_str(token.as_str()),
+                Rule::string_escape => match token.as_str().chars().nth(1).unwrap() {
+                    '"' => string.push('"'),
+                    '\\' => string.push('\\'),
+                    'n' => string.push('\n'),
+                    'r' => string.push('\r'),
+                    't' => string.push('\t'),
+                    _ => unreachable!(),
+                },
```

Comment:
The way that the `pest` library works, at the point we're doing the match we know that the input was valid for the `string_escape` rule, which is defined as follows:
```pest
string_escape  = @{ "\\" ~ ("\"" | "\\" | "n" | "r" | "t") }
```

So that case is indeed unreachable. This is similar to the other uses of `unreachable` in the parser and an idiomatic use of `pest`.

Reply:
When these diverse we have a bug. A parse error would indicate that there's something wrong with the input, when in truth the error is in the code. So I think the panic is appropriate.